pkg/prometheus/operator.go

```diff
@@ -1075,7 +1075,7 @@ func (c *Operator) selectServiceMonitors(p *monitoringv1.Prometheus) (map[string
 	}
 
 	// If 'ServiceMonitorNamespaceSelector' is empty, only check own namespace.
```

Comment:
With this change I find this comment a bit misleading. Would you mind changing it to something like:

```go
// If 'ServiceMonitorNamespaceSelector' is nil (yaml, json: null), only check own namespace. If it is empty, check all namespaces.
```

Same would need to be copied here:
https://github.com/coreos/prometheus-operator/blob/master/pkg/client/monitoring/v1/types.go#L63-L64